packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -390,18 +409,20 @@ function shallowCompatibilityTest(
 		return data[typeNameSymbol] === schema.identifier;
 	}
 
+	// Array schemas strictly allow array data as input
 	if (isReadonlyArray(data)) {
 		return schema.kind === NodeKind.Array;
 	}
 	if (schema.kind === NodeKind.Array) {
 		return false;
 	}
 
+	// Map schemas allow TypeScript maps and Record-like objects as input
 	if (data instanceof Map) {
 		return schema.kind === NodeKind.Map;
 	}
 	if (schema.kind === NodeKind.Map) {
-		return false;
+		return typeof data === "object" && data !== null;
```

Comment:
This does not align with map's nodes inputs of `Iterable<[string, InsertableTreeNodeFromImplicitAllowedTypes<T>]>` (see SchemaFactory.map).